tests/test_upload.py

```diff
@@ -286,3 +286,39 @@ def _wait_for_upload(procs, total):
     for proc in process_mocks:
         proc.start.assert_called()
         proc.stop.assert_called()
+
+
+def test_compose_file_mapping(tmpdir):
+    local_map = {
+        tmpdir.join("local.path"): "test/gcs.1a",
+        tmpdir.join("test.dir"): "test/gcs.1b"
+    }
+    input_map = {
+        "gs://bucket/gcs.path": "test/gcs.2",
+    }
```

Comment:
Looks like these examples are missing a test for square bracket removal?

Reply:
add in next PR